packages/common/src/api/tan-query/comments.ts

```diff
@@ -0,0 +1,719 @@
+        [QUERY_KEYS.comment, commentId],
+        (prevData: CommentOrReply | undefined) =>
+          ({
+            ...prevData,
+            isEdited: true,
+            message: newMessage
+          } as CommentOrReply)
+      )
+      return { prevComment }
+    },
+    onError: (error: Error, args, context) => {
+      const { commentId } = args
+      reportToSentry({
+        error,
+        additionalInfo: args,
+        name: 'Comments'
+      })
+      // Toast standard error message
+      dispatch(toast({ content: messages.mutationError('editing') }))
+
+      // Note: context could be undefined if the onMutate threw before returning
+      if (context) {
+        const { prevComment } = context
+        // Revert our optimistic cache change
+        queryClient.setQueryData(
+          [QUERY_KEYS.comment, commentId],
+          (prevData: CommentOrReply | undefined) =>
+            ({
+              ...prevData,
+              // NOTE: intentionally only reverting the pieces we changed in case another mutation happened in between this mutation start->error
+              isEdited: prevComment?.isEdited,
+              message: prevComment?.message
+            } as CommentOrReply)
+        )
+      }
+    }
+  })
+}
+
+type ReportCommentArgs = {
+  commentId: ID
+  userId: ID
+  trackId: ID
+  currentSort: CommentSortMethod
+}
+export const useReportComment = () => {
+  const { audiusSdk, reportToSentry } = useAudiusQueryContext()
+  const queryClient = useQueryClient()
+  const dispatch = useDispatch()
+  return useMutation({
+    mutationFn: async ({ userId, commentId }: ReportCommentArgs) => {
+      const sdk = await audiusSdk()
+      await sdk.comments.reportComment(userId, commentId)
+    },
+    onMutate: ({ trackId, commentId, currentSort }) => {
+      // Optimistic update - filter out the comment
+      queryClient.setQueryData<InfiniteData<ID[]>>(
+        [QUERY_KEYS.trackCommentList, trackId, currentSort],
+        (prevData) => {
+          if (!prevData) return
+          const newState = cloneDeep(prevData)
+          // Filter out our reported comment
+          newState.pages = newState.pages.map((page) =>
+            page.filter((id) => id !== commentId)
+          )
+          return newState
+        }
+      )
+      queryClient.resetQueries([QUERY_KEYS.comment, commentId])
+    },
+    onError: (error: Error, args) => {
+      const { trackId, currentSort } = args
+      reportToSentry({
+        error,
+        additionalInfo: args,
+        name: 'Comments'
+      })
+      // Generic toast error
+      dispatch(toast({ content: messages.mutationError('reporting') }))
+
+      // Reload data
+      queryClient.resetQueries([
+        QUERY_KEYS.trackCommentList,
+        trackId,
+        currentSort
+      ])
+    }
+  })
+}
+
+type MuteUserArgs = {
+  mutedUserId: ID
+  userId: ID
+  isMuted: boolean
+  trackId?: ID
+  currentSort?: CommentSortMethod
+}
+
+export const useMuteUser = () => {
```

Comment:
@isaacsolo I ported this over from a-query and tested it out, but do take a peek and see if anything looks off